ArduCopter/mode_guided.cpp

```diff
@@ -51,39 +51,47 @@ bool ModeGuided::init(bool ignore_checks)
 // do_user_takeoff_start - initialises waypoint controller to implement take-off
 bool ModeGuided::do_user_takeoff_start(float takeoff_alt_cm)
 {
-    guided_mode = Guided_TakeOff;
-
-    // initialise wpnav destination
-    Location target_loc = copter.current_loc;
-    Location::AltFrame frame = Location::AltFrame::ABOVE_HOME;
-    if (wp_nav->rangefinder_used_and_healthy() &&
-            wp_nav->get_terrain_source() == AC_WPNav::TerrainSource::TERRAIN_FROM_RANGEFINDER &&
-            takeoff_alt_cm < copter.rangefinder.max_distance_cm_orient(ROTATION_PITCH_270)) {
-        // can't takeoff downwards
-        if (takeoff_alt_cm <= copter.rangefinder_state.alt_cm) {
-            return false;
+
+    if (motors->get_spool_state() != AP_Motors::SpoolState::THROTTLE_UNLIMITED) {
+        guided_mode = Guided_PreTakeOff;
+        _takeoff_alt_cm = takeoff_alt_cm;
+        return true;
+    } else {
+
+        guided_mode = Guided_TakeOff;
+
+        // initialise wpnav destination
+        Location target_loc = copter.current_loc;
+        Location::AltFrame frame = Location::AltFrame::ABOVE_HOME;
+        if (wp_nav->rangefinder_used_and_healthy() &&
+                wp_nav->get_terrain_source() == AC_WPNav::TerrainSource::TERRAIN_FROM_RANGEFINDER &&
+                takeoff_alt_cm < copter.rangefinder.max_distance_cm_orient(ROTATION_PITCH_270)) {
+            // can't takeoff downwards
+            if (takeoff_alt_cm <= copter.rangefinder_state.alt_cm) {
+                return false;
```

Comment:
I think you need to add `copter.set_auto_armed(false);` before returning false here so the heli actually spools down.